node/src/components/linear_chain_sync.rs

```diff
@@ -301,13 +301,10 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
             State::SyncingDescendants {
                 ref current_block, ..
             } => {
-                match current_block.as_ref() {
-                    Some(expected) => assert_eq!(
-                        expected, &block_header,
-                        "Block execution result doesn't match received block."
-                    ),
-                    None => panic!("Unexpected block execution results."),
-                }
+                assert_eq!(
+                    **current_block, block_header,
+                    "Block execution result doesn't match received block."
+                );
```

Comment:
I don't understand how this would work – when we transition from `SyncingTrustedHash` to `SyncingDescendants`, we use trusted hash block header as `current_block`. Then, we download its child and eventually `block_handled` is called with that child, but here we compare `current_block` (trusted block header) with the `block_header` that should be its direct child. I know I wrote this code and I should know better but I don't see it.

Reply:
I just tried running it and it works. I don't know what I am missing.